pages/members/[id]/index.js

```diff
@@ -1,10 +1,28 @@
+import { getDataURL, getImgURL } from 'helper-functions/urls';
+import PropTypes from 'prop-types';
+import fetch from 'cross-fetch';
 import React from 'react';
-import { useRouter } from 'next/router';
 import Profile from 'components/member-profile';
 
-const MemberProfile = () => {
-  let { query } = useRouter();
-  return <Profile id={query.id} />;
+const MemberProfile = ({ imageLink, data }) => {
+  return <Profile imageLink={imageLink} membersData={data} />;
+};
+
+export async function getServerSideProps(context) {
+  const imgLink = getImgURL(context.params.id);
+  const jsonUrl = getDataURL(context.params.id);
+  const res = await fetch(jsonUrl);
+  const returnData = await res.json();
+
+  return { props: { imageLink: imgLink, data: returnData } };
+}
+
+MemberProfile.propTypes = {
+  imageLink: PropTypes.string
+};
+
+MemberProfile.propTypes = {
+  data: PropTypes.object
```

Comment:
Why are here two propTypes? Both `imageLink` and `data` can be defined inside one. 
Also, defaultProps are missing.